File: 2019/Aug(81)/lecture_002_array/ArrayBasic.java
public class ArrayBasic {

    static void fun(int[] arr) {
        for (int i = 0; i < arr.length; i++) {
            arr[i] = i + 1;
        }
    }

     public static void main(String[] args) {
        int[] arr = new int[10];

        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }

        System.out.println();
        fun(arr);

        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }

    }

}

File: 2019/Aug(81)/lecture_002_array/ArrayBasic.cpp
#include <iostream>
#include <vector>

using namespace std;

void fun(vector<int>& arr)
{
    for (int i = 0; i < arr.size(); i++)
    {
        arr[i] = i + 1;
    }
}

int main(int args, char **argv)
{
    vector<int> arr(10, 0);

    for (int i = 0; i < arr.size(); i++)
    {
        cout << arr[i] << " ";
    }

    cout << endl;
    fun(arr);
    
    for (int i = 0; i < arr.size(); i++)
    {
        cout << arr[i] << " ";
    }

    return 0;
}